packages/document/docs/en/api/javascript-api/instance.mdx

```diff
@@ -267,6 +267,119 @@ await rsbuild.startDevServer({
 });
 ```
 
+## rsbuild.createDevServer
+
+Rsbuild comes with a built-in dev server designed to improve the development experience. When you run the `rsbuild dev` command, the server will start, providing features such as page preview, routing, and hot module reloading.
+
+If you want to integrate Rsbuild dev server into an custom server, you can use this method to get the instance methods of Rsbuild dev server and call them on demand.
+
+- **Type:**
+
+```ts
+type RsbuildDevServer = {
+  /** start the Rsbuild DevServer */
+  listen: () => Promise<{
+    urls: string[];
+    port: number;
+    server: Server;
+  }>;
+
+  /** The following APIs will be used when you use a custom server */
+
+  /**
+   * The resolved rsbuild server config
+   *
+   * More config can be get from `rsbuild.getNormalizedConfig` api
+   */
+  config: {
```

Comment:
how about just export `port` instead of `config.port`, since most config are obtained by `getNormalizedConfig`.